Brokerages/GDAX/GDAXBrokerage.Messaging.cs

```diff
@@ -336,8 +336,8 @@ private void OrderMatch(string data)
                 ? OrderStatus.Filled : OrderStatus.PartiallyFilled;
 
             OrderDirection direction;
-            // Messages are always from the perspective of the market maker. Flip it in cases of a market order.
-            if (order.Type == OrderType.Market)
+            // Messages are always from the perspective of the market maker. Flip direction if executed as a taker.
+            if (order.BrokerId[0] == message.TakerOrderId)
```

Comment:
Any concern with `BrokerId` being empty here? What about multiple broker ids? Do we need to check `order.BrokerId.Any(bid => bid == message.TakerOrderId` ?

Reply:
I believe with GDAX there will always only be one element in order.BrokerId, the empty case should already be handled in line 317.